cmd/node-agent/main.go

```diff
@@ -0,0 +1,220 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-systemd/dbus"
+	"github.com/coreos/go-systemd/unit"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	semver "k8s.io/kubernetes/pkg/version"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	myPodNameEnvName               = "MY_POD_NAME"
+	desiredVersionAnnotation       = "updates.coreos.com/desired-version"
+	currentVersionAnnotation       = "updates.coreos.com/current-version"
```

Comment:
I feel like we should be retrieving this from the source, rather than double storing in an annotation.

In the case of the node-object,  `node.Status.NodeInfo.KubeletVersion` would give us the current running version, but that could be off with self-hosted kubelet. For example, on-host is v1.1.1, but self-hosted is v1.1.2 -- then the node object would report 1.1.2 (and we would fail to update the on-host copy).

So in this case, I think we want to determine the on-host version (from the source). This makes me thing storing the actual version in a systemd drop-in might not be ideal (and instead just have the kubelet.service file source a file that is more easily parsed with probably just `KEY=value` pairs.